packages/web/src/scss/themes/index.scss

```diff
@@ -0,0 +1,25 @@
+@use 'sass:map';
+@use 'sass:string';
+@use '@themes' as themes;
+@use '../tools/themes' as themes-tools;
+
+$is-default-theme: true;
+$global-prefix: 'spirit-';
+
+@function -singularize($string) {
+    @return string.slice($string, 1, string.length($string) - 1);
+}
+
+@each $key, $value in themes.$themes {
+    $selector: if($is-default-theme, ':root, .#{$global-prefix}#{$key}', '.#{$global-prefix}#{$key}');
+
+    #{$selector} {
+        @each $token-type-key, $token-type-value in $value {
+            $token-type-name: -singularize($token-type-key);
```

Comment:
How about moving this logic to the tools so it can be tested too?

I'd be insterested in testing of:

- generated theme selectors,
- the singularization stuff.

Reply:
Singularization could be placed in `tools/_string.scss`.